### PR TITLE
Show the name of inventory item when selection changes

### PR DESCRIPTION
**Pull request by Zeno-**

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -3853,25 +3851,39 @@ void Game::updateFrame(std::vector<aabb3f> &highlight_boxes,
 		Inventory
 	*/
 
-	if (client->getPlayerItem() != runData->new_playeritem)
+	bool inventory_selection_changed = false;
+	bool inventory_changed = false;
```
Comment (SmallJoker) [inline src/game.cpp]: The type `bool` is set to `false` by default.

Comment (est31) [inline src/game.cpp]: I wouldn't be that sure. [section 7.6.8 paragraph 10](http://www.open-std.org/JTC1/SC22/WG14/www/docs/n1256.pdf#page=138&zoom=auto,-27,816) of the C standard handles objects with "automatic storage duration" (like this one), and states that values of such uninitialized objects are indeterminate.

Comment (Zeno-) [inline src/game.cpp]: est31 is correct.

The only objects that are initialized (to 0) are globals and static locals (these are not the terms the standard uses, but it's close enough)

Comment (nerzhul) [inline src/game.cpp]: It's better to be sure that the variable is initialized correctly than reporting to standard which could not be correctly followed by some non-free OSes :)

**Comment by C1ffisme** [comment]
:+1:

**Comment by est31** [comment]
Found three issues:

1. if you join, the item's name isn't shown. (perhaps that doesn't need fixing, I'm not sure)
2. if you pick up an item, it doesn't show up (weird, because the code states otherwise)
3. perhaps give mods a method to find out whether the engine already shows the item? Because in dreambuilder a mod already does right this, so perhaps it can disable it being shown when the client actually shows it (and not use some broken method based on the version). This fits into @kwolekr's changes to allow per-user HUD.

**Comment by Zeno-** [comment]
Hmm, Ok. I tested but didn't experience those issues you mentioned (weird). I'll review it.

I don't think #1 is an issue...

**Comment by est31** [comment]
I've tried with that log message uncommented.When you pick up stuff, the wield item does get updated. Like you, I also have those update events, approximately all 10 seconds.

**Comment by Zeno-** [comment]
@est31 If it wasn't for this behaviour, item #2 that you mentioned a bug is very easy to fix (and item #1 actually) -- simply display the text when wield item changes. But, seeing the text display every 2-10 seconds is kind of annoying, naturally. The other related issue is that `client->getLocalInventoryUpdated()` returns true when you dig (because dirt or whatever gets added to the inventory, tool wear etc), causing the wield item to be changed even if the affected item isn't the one being wielded. I've been wanting to look at the cause of this somewhat undesirable behaviour but not really had access to a computer for long enough. I'll be back from my trip tomorrow so can look at it more then

**Pull request closed without merging**